tools/ci_connector_ops/ci_connector_ops/pipelines/contexts.py

```diff
@@ -404,23 +418,106 @@ async def __aexit__(
                 self.logger.error("Uploading the report to S3 failed.")
         await asyncify(update_commit_status_check)(**self.github_commit_status)
         if self.should_send_slack_message:
-            await asyncify(send_message_to_webhook)(self.create_slack_message(), self.reporting_slack_channel)
+            await asyncify(send_message_to_webhook)(self.create_slack_message(), self.reporting_slack_channel, self.slack_webhook)
         # Supress the exception if any
         return True
 
     def create_slack_message(self) -> str:
-        message = f"*<{self.gha_workflow_run_url}|{self.pipeline_name}>*\n"
+        raise NotImplementedError
+
+
+class PublishConnectorContext(ConnectorContext):
```

Comment:
Declare a new subclass is helpful here because:
- I can make the context aware if its a pre-release publish or not and template the slack message accordingly
- I can propagate secrets received from CLI options (docker hub, gcs credentials etc..) or env var to Step via context attribute. Env var as passed in to the command entrypoint as (required) Click options so I can guarantee their existence at runtime.